lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -43,7 +44,47 @@ function runA11yChecks() {
       'blink': {enabled: false},
       'server-side-image-map': {enabled: false}
     }
+  }).then(axeResult => {
+    // Augment the node objects with outerHTML snippet & custom path string
+    axeResult.violations.forEach(v => v.nodes.forEach(node => {
+      node.path = getNodePath(node.element);
+      node.snippet = getOuterHTMLSnippet(node.element);
+    }));
+
+    return axeResult;
   });
+
+  // Adapated from DevTools' SDK.DOMNode.prototype.path
+  //   https://github.com/ChromeDevTools/devtools-frontend/blob/7a2e162ddefd/front_end/sdk/DOMModel.js#L530-L552
+  // TODO: Doesn't handle frames or shadow roots...
```

Comment:
Can you file a bug on this? We have some other "handle shadow root" updates to make in the codebase. We can address them all after io.